### ISSUE TITLE
Getting rid of handlers

### ISSUE THREAD

**Issue by halvors** [InfectedLan/InfectedAPI#141]
Getting rid of the awful handlers have been a long reach goal. Today i came up with and kindof abstraction idea.

What i propose that we do is that we add a new object called DatabaseUser which is a representation of all the database related user stuff. There we add all the functions that include SQL code.

Then we make User extend DatabaseUser. Thereby eliminate the need for the UserHandler.
Off course we will need some static functions in DatabaseUser to initially deal with registration etc.

**Comment by petterroea** [comment]
This is practically renamind UserHandler to DatabaseUser, except having the methods in the same class on a theoretical level.

My problem with handles is having so_many_files open because each class has its own handler class for it. The optimal would be to cut it down to one file per class.

**Comment by halvors** [comment]
Even if we have User and DatabaseUser we still have one file per class you know?...

**Comment by petterroea** [comment]
Re-read what i wrote

"My problem with handles is having so_many_files open because each class has its own handler class for it."

With "one file per class" i meant that i dont want two files(where one has a helper/handler class). I want one file, one class. No helpers. No database abstractions. Just the class itself. No helpers. No database loaders. Just the class itself, compeltely self-sustained without the need of a helper or database abstraction, in one file.

**Comment by halvors** [comment]
The thing here is that we should keep static methods and instance stuff separeted.

**Comment by petterroea** [comment]
Yeah, but at the same time, having everything for an object in that class makes things simpler. It all boils down to the fact that i don't want to edit two files when i create objects. This isn't c++.

**Comment by halvors** [comment]
Objects and Handlers have 2 pretty different uses. A object doesn't need to have a handler, and a handler doesn't need a object. You can have a handler interfacing with other objects than the one it's the handler for...

**Comment by halvors** [comment]
Should we close this?

**Comment by petterroea** [comment]
Nah. we just need time to find a proper implementation

**Comment by petterroea** [comment]
We ARE doing some changes to how database works, so we could get away with doing this at the same time.

I want more test coverage tho.
